src/storage/src/sink/healthcheck.rs

```diff
@@ -147,7 +140,8 @@ impl Healthchecker {
                         // Sync to the new upper, go to the loop again
                         self.sync(&actual_upper.0).await;
```

Comment:
FWIW I think we can plausibly now just set `self.upper = actual_upper` here directly and drop the `sync` /  `listener` stuff. Should further reduce the load on persist, and I'm not sure we're getting a ton out of it... loading a lot of data, and the CAS loop will already handle a stale upper correctly.